pkg/workflow/argo.go

```diff
@@ -199,6 +201,70 @@ func (a *argo) Submit(ctx context.Context, l logr.Logger, g *v1alpha1.Applicatio
 	return nil
 }
 
+func (a *argo) GenerateReversal(ctx context.Context, l logr.Logger, nodes map[string]v1alpha12.NodeStatus, g *v1alpha1.ApplicationGroup) error {
+	if g == nil {
+		l.Error(nil, "ApplicationGroup object cannot be nil")
+		return fmt.Errorf("applicationGroup object cannot be nil")
+	}
+
+	initWorkflowObject(&a.rwf)
+
+	// Set name and namespace based on the input application group
+	a.rwf.Name = fmt.Sprintf("%s-reversal", g.Name)
+	a.rwf.Namespace = workflowNamespace()
+
+	err := a.generateReversalWorkflow(ctx, l, nodes, g)
+	if err != nil {
+		l.Error(err, "failed to generate reversal workflow")
+		return fmt.Errorf("failed to generate argo reversal workflow : %w", err)
+	}
+
+	return nil
+}
+
+func (a *argo) SubmitReversal(ctx context.Context, l logr.Logger, g *v1alpha1.ApplicationGroup) error {
+	if a.rwf == nil {
+		l.Error(nil, "reversal workflow object cannot be nil")
+		return fmt.Errorf("reversal workflow object cannot be nil")
+	}
+
+	if g == nil {
+		l.Error(nil, "applicationGroup object cannot be nil")
+		return fmt.Errorf("applicationGroup object cannot be nil")
+	}
+
+	obj := &v1alpha12.Workflow{
+		ObjectMeta: v1.ObjectMeta{Labels: make(map[string]string)},
+	}
+
+	err := a.cli.Get(ctx, types.NamespacedName{Namespace: a.rwf.Namespace, Name: a.rwf.Name}, obj)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Add OwnershipReference
+			err = controllerutil.SetControllerReference(g, a.rwf, a.scheme)
```

Comment:
Like we discussed - instead of setting the ownership reference to the applicationgroup we can make it a child of the forward workflow (`a.wf`)